pkg/api/status/v1.go

```diff
@@ -0,0 +1,110 @@
+package status
+
+import (
+	"fmt"
+	"math"
+	"net/http"
+
+	"github.com/thanos-io/thanos/pkg/api"
+
+	"github.com/go-kit/log"
+	"github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/route"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/tsdb"
+	"github.com/prometheus/prometheus/tsdb/index"
+	v1 "github.com/prometheus/prometheus/web/api/v1"
+	extpromhttp "github.com/thanos-io/thanos/pkg/extprom/http"
+	"github.com/thanos-io/thanos/pkg/logging"
+)
+
+// Stat holds the information about individual cardinality.
+type Stat struct {
+	Name  string `json:"name"`
+	Value uint64 `json:"value"`
+}
+
+func convertStats(stats []index.Stat) []Stat {
+	result := make([]Stat, 0, len(stats))
+	for _, item := range stats {
+		item := Stat{Name: item.Name, Value: item.Count}
+		result = append(result, item)
+	}
+	return result
+}
+
+// TSDBStatus has information of cardinality statistics from postings.
+type TSDBStatus struct {
```

Comment:
We could remove these structs if https://github.com/prometheus/prometheus/pull/10783 gets merged.

Reply:
It was merged, we can re-use the struct now :+1:

Reply:
Do I need to go get the latest prometheus version from master/main, or do we need to wait for a new point release? I'm not sure how Thanos manages dependencies.

Reply:
We can use latest main directly.

Reply:
Yes, we need to update Cortex or Cortex folks will update Thanos when they need to

Reply:
One more thing, what is the reason for using github.com/akanshat/cortex instead github.com/cortexproject/cortex? Which one should actually be updated?

Reply:
I think we use it for groupcache changes https://github.com/thanos-io/thanos/pull/4818. Maybe Thanos version in fork can be updated cc: @akanshat?

Reply:
Yes, so while adding groupcache we had to make some changes in Thanos, which affected cortex imports just a little. Then we decided to update my fork of cortex and point Thanos to it, and once groupcache changes were merged we were going to create a PR to Cortex, to have those changes merged and point Thanos back to the original cortex.

Now [#4651](https://github.com/cortexproject/cortex/pull/4651) is open, it's been reviewed by @GiedriusS and @bwplotka but not by any cortex maintainer yet. I am not sure what the next step is. @GiedriusS would you like to help me out with this?

Reply:
While waiting for that to get merged, should we maybe rebase the fork against the latest master in Cortex in order to get this PR unblocked? Otherwise, we will blocked whenever we try to update the prometheus/prometheus dependency.

Reply:
No need for relase

Reply:
We could rebase cortex fork.